### ISSUE TITLE
`reproject` has Locus issues

### ISSUE THREAD

**Issue by rafaqz** [rafaqz/Rasters.jl#829]
It seems to give slightly different results if `shiftlocus(Center, rast)` is used before or after `reproject`. This could be intrinsically from shiftlocus being weird on an Irregular lookup. But it needs investigation.

**Comment by asinghvi17** [comment]
mwe?

**Comment by asinghvi17** [comment]
Just updated the example btw

**Comment by rafaqz** [comment]
Yeah. I guess for `Intervals` we should do the interval bounds properly and make them `Explicit` then do the center as well separately.

**Comment by asinghvi17** [comment]
Hmm that is only necessary if the intervals are Center I think.  For start or end, since the transform is invertible we already get this for free with irregular but implicit intervals.

**Comment by rafaqz** [comment]
Yeah the center moving is the problem. I guess instead it could just never return Center intervals

**Comment by asinghvi17** [comment]
This is also a nice candidate for bin edges style explicit...

**Comment by rafaqz** [comment]
Totally, still a bit weird that the centre won't be in the center
